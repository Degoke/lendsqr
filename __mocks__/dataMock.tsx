import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
