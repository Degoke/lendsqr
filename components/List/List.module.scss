.list {
  margin: 1rem 0;

  p {
    padding: 0 1.5rem;
    text-transform: capitalize;
    margin-top: 1.5rem;

    @media all and (max-width: 800px) {
      display: none;
    }
  }
}
