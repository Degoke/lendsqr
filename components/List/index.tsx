import { IListItem } from '../../utils/types/list'
import ListItem from '../ListItem'
import styles from './List.module.scss'

type Props = {
  items: IListItem[]
  title: string
}

export default function List({ items, title }: Props) {
  return (
    <div className={styles.list}>
      <p>{title}</p>
      <ul>
        {items.map((item) => (
          <ListItem key={item.text} item={item} />
        ))}
      </ul>
    </div>
  )
}
