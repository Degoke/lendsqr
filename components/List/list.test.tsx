import { render, screen } from '@testing-library/react'
import List from './index'
import { mockCard, mockImage, mockList } from '../../__mocks__/dataMock'

describe('List tests', () => {
  it('renders List component unchanged', () => {
    const { container } = render(
      <List items={mockList.items} title={mockList.title} />
    )
    expect(container).toMatchSnapshot()
  })
})
