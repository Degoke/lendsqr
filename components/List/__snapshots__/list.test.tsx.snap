// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List tests renders List component unchanged 1`] = `
<div>
  <div
    class="list"
  >
    <p>
      settings
    </p>
    <ul>
      <li>
        <a
          class="item current"
          href=""
        >
          <img
            alt="preferences nav button"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <span>
            preferences
          </span>
        </a>
      </li>
      <li>
        <a
          class="item current"
          href=""
        >
          <img
            alt="fees and pricing nav button"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <span>
            fees and pricing
          </span>
        </a>
      </li>
      <li>
        <a
          class="item current"
          href=""
        >
          <img
            alt="audit logs nav button"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <span>
            audit logs
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
